src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/AssessmentResourceIntTest.kt

```diff
@@ -127,6 +145,147 @@ class AssessmentResourceIntTest : SqsIntegrationTestBase() {
     }
   }
 
+  @Nested
+  open inner class PostponeCase {
+
+    private inner class PostponeCaseRequestNoAgent(
+      val reasonTypes: LinkedHashSet<PostponeCaseReasonType> = LinkedHashSet(),
+    )
+
+    private val anPostponeCaseRequestNoAgent = PostponeCaseRequestNoAgent(
+      reasonTypes = LinkedHashSet(listOf(PostponeCaseReasonType.ON_REMAND)),
+    )
+
+    private val anPostponeCaseRequestWithNoReason = PostponeCaseRequest(
+      reasonTypes = LinkedHashSet(),
+      agent = Agent("a user", PRISON_CA, "ABC"),
+    )
+
+    private val anPostponeCaseRequest = PostponeCaseRequest(
+      reasonTypes =
+      LinkedHashSet(
+        listOf(
+          PostponeCaseReasonType.ON_REMAND,
+          PostponeCaseReasonType.COMMITED_OFFENCE_REFERRED_TO_LAW_ENF_AGENCY,
+        ),
+      ),
+      agent = Agent("a user", PRISON_CA, "ABC"),
+    )
+
+    @Test
+    fun `should return unauthorized if no token`() {
+      webTestClient.put()
+        .uri(POSTPONE_ASSESSMENT_URL)
+        .bodyValue(anPostponeCaseRequest)
+        .exchange()
+        .expectStatus()
+        .isUnauthorized
+    }
+
+    @Test
+    fun `should return forbidden if no role`() {
+      webTestClient.put()
+        .uri(POSTPONE_ASSESSMENT_URL)
+        .headers(setAuthorisation())
+        .bodyValue(anPostponeCaseRequest)
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `should return forbidden if wrong role`() {
+      webTestClient.put()
+        .uri(POSTPONE_ASSESSMENT_URL)
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+        .bodyValue(anPostponeCaseRequest)
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Sql(
+      "classpath:test_data/reset.sql",
+      "classpath:test_data/an-passed-pre-release-checks-for-postponement.sql",
+    )
+    @Test
+    fun `should postpone an offenders assessment`() {
+      // When
```

Comment:
I've only ever seen people use given / when / then 

`given` some initial state
`when` this happens
`then` this is the result

Reply:
got that the wrong way round, will change